src/middlewared/middlewared/plugins/kubernetes_linux/update.py

```diff
@@ -142,8 +142,15 @@ async def validate_data(self, data, schema, old_data):
             else:
                 verrors.add(f'{schema}.cluster_dns_ip', 'Please specify cluster_dns_ip.')
 
-        if data['pool'] and not await self.middleware.call('pool.query', [['name', '=', data['pool']]]):
-            verrors.add(f'{schema}.pool', 'Please provide a valid pool configured in the system.')
+        if data['pool']:
+            pool_details = await self.middleware.call('pool.query', [['name', '=', data['pool']]])
+            if not pool_details:
+                verrors.add(f'{schema}.pool', 'Please provide a valid pool configured in the system.')
+            elif await self.middleware.call('pool.get_usb_disks', pool_details[0]['id']):
+                verrors.add(
+                    f'{schema}.pool',
+                    f'Applications cannot be configured on {data["pool"]!r} pool as it consumes USB based disks.'
```

Comment:
Since this returns an array of disks, probably would be nice to put them in the validation error so end user can see them.